src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -503,3 +510,174 @@ def _sql_alter_from(self, table: Table, target_table_key: str, ws_id: int):
+        migration_status_refresher: MigrationStatusRefresher,
+        principal_acl: PrincipalACL,
+    ):
+        self._tc = tables_crawler
+        self._gc = grant_crawler
+        self._backend = backend
+        self._workspace_info = workspace_info
+        self._group = group_manager
+        self._migration_status_refresher = migration_status_refresher
+        self._principal_acl = principal_acl
+
+    def migrate_acls(
+        self,
+        *,
+        target_catalog: str | None = None,
+        legacy_table_acl: bool = True,
+        principal: bool = True,
+        hms_fed: bool = False,
+    ) -> None:
+        acl_strategies = []
+        workspace_name = self._workspace_info.current()
+        if legacy_table_acl:
+            acl_strategies.append(AclMigrationWhat.LEGACY_TACL)
+        if principal:
+            acl_strategies.append(AclMigrationWhat.PRINCIPAL)
+        all_grants_to_migrate = self._gc.snapshot()
+        all_migrated_groups = self._group.snapshot()
+        all_principal_grants = self._principal_acl.get_interactive_cluster_grants()
+        tables = self._tc.snapshot()
+
+        if not tables:
+            logger.info("No tables found to acl")
+            return
+        if hms_fed:
+            tables_to_migrate = self._get_hms_fed_tables(tables, target_catalog if target_catalog else workspace_name)
+        else:
+            tables_to_migrate = self._get_migrated_tables(tables)
+
+        self._migrate_acls(
+            acl_strategies,
+            all_grants_to_migrate,
+            all_migrated_groups,
+            all_principal_grants,
+            tables_to_migrate,
+        )
+
+    def _get_migrated_tables(
+        self,
+        tables: list[Table],
+    ):
+        # gets all the migrated table to apply ACLs to
+        tables_to_migrate = []
+        seen_tables = self._migration_status_refresher.get_seen_tables()
+        reverse_seen_tables = {v: k for k, v in seen_tables.items()}
+        for table in tables:
+            if table.key not in reverse_seen_tables:
+                logger.warning(f"Table {table.key} not found in migration status. Skipping.")
+                continue
+            dst_table_parts = reverse_seen_tables[table.key].split(".")
+            if len(dst_table_parts) != 3:
+                logger.warning(
+                    f"Invalid table name {reverse_seen_tables[table.key]} found in migration status. Skipping."
+                )
+                continue
+            rule = Rule(
+                self._workspace_info.current(),
+                dst_table_parts[0],
+                table.database,
+                dst_table_parts[1],
+                table.name,
+                dst_table_parts[2],
+            )
+            table_to_migrate = TableToMigrate(table, rule)
+            tables_to_migrate.append(table_to_migrate)
+        return tables_to_migrate
+
+    def _get_hms_fed_tables(self, tables: list[Table], target_catalog):
```

Comment:
```suggestion
    def _get_hms_fed_tables(self, tables: list[Table], target_catalog: str) -> list[TableToMigrate]:
```
add return type so that mypy finds more bugs for us.

Reply:
Done